distributed/shuffle/_shuffle_extension.py

```diff
@@ -434,6 +501,7 @@ async def _get_shuffle(
 
                 raise Reschedule()
             else:
+                self.raise_if_closed()
```

Comment:
Feels a bit odd to fail here. I'd expect this to happen somewhere else (and all tests pass if I remove this).

Reply:
IIUC, we technically need this here. Otherwise we can run into a very unlikely edge case where the extension has been closed since we started sending the RPC to the scheduler. This would leave the new `Shuffle` instance open indefinitely. We can likely relax this condition once we have proper cleanup upon task release implemented. Tests are expected to pass since I have not written once for this specific scenario and most (all?) dealing with closing workers expect the shuffle to have started before anything happens. Now that you bring it up though, this _might_ be worth another test if we want to make this rock-solid.